provider/resources.go

```diff
@@ -2840,6 +2840,38 @@ func Provider() *tfbridge.ProviderInfo {
 						Type:     "string",
 						AltTypes: []tokens.Type{awsType(rdsMod, "StorageType", "StorageType")},
 					},
+					"name": {
+						Name: func() string {
+							// We inject `name` into the underlying provider so it shows
+							// up in the schema. It is never observed non-nil by the
+							// upstream provider and we warn when you set it.
+
+							p.ResourcesMap().Get("aws_db_instance").Schema().
+								Set("name", shimv2.NewSchema(&tfschema.Schema{
+									Type:     tfschema.TypeString,
+									Optional: true,
+								}))
+							return "name"
+						}(),
+						DeprecationMessage: "This property has been deprecated. " +
+							"Please use 'dbName' instead.",
+					},
+				},
+				PreCheckCallback: func(
+					ctx context.Context, config resource.PropertyMap, meta resource.PropertyMap,
+				) (resource.PropertyMap, error) {
+					if name, ok := config["name"]; ok {
+						// Both `name` and `dbName` are set, so error
+						if _, ok := config["dbName"]; ok {
+							return nil, fmt.Errorf("Cannot specify both name and dbName, " +
+								"please set only dbName")
+						}
+						// Name doesn't actually exist on the underlying provider anymore,
```

Comment:
Question, don't we need to emit a deprecation warning here where we do `config["dbName"] = name`? Or something else will emit a deprecation warning anyway?

Reply:
We need to emit the warning, but we don't have a way to do so here. https://github.com/pulumi/pulumi-terraform-bridge/pull/1333 provides the logging context to `PreCheckCallback`. When that merges, we add this patch:

```patch
 provider/resources.go | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/provider/resources.go b/provider/resources.go
index 9706f52d0d..e44a569b3f 100644
--- a/provider/resources.go
+++ b/provider/resources.go
@@ -2866,6 +2866,8 @@ func Provider() *tfbridge.ProviderInfo {
 							return nil, fmt.Errorf("Cannot specify both name and dbName, " +
 								"please set only dbName")
 						}
+						tfbridge.GetLogger(ctx).Warn("The `name` property is deprecated, " +
+							"please use `dbName` instead")
 						// Name doesn't actually exist on the underlying provider anymore,
 						// so we make sure it only sees `dbName`, not `name`.
 						config["dbName"] = name
```

Reply:
Well you can send a warning to glog  or even printf "WARNING" that should work already AFAIK, even if not ideal (not tied to the resource). 

How does the experience look like right now? We need this for go/no-go.

https://github.com/pulumi/pulumi-terraform-bridge/pull/1333 needs tests and some iteration it seems like, trying to weight in if we have enough time to land that first.